corehq/apps/app_manager/app_translations/parser.py

```diff
@@ -60,12 +60,31 @@ def _add_module_and_form_sheet(self, ws, po_entries):
             _type, _sheet_name, _unique_id = re.match(context_regex, context).groups()
             ws.append([_type, _sheet_name, po_entry.msgid, po_entry.msgstr, _unique_id])
 
+    @staticmethod
+    def _get_rows_for_module_sheet(consolidated_po_entries):
+        """
+        get po_entries listed according to their expected order in the final set
+        by using occurrences
+        """
+        rows = {}
+        po_entries = []
+        # align entries in a dict with their index as the key
+        for po_entry in consolidated_po_entries:
+            occurrences = po_entry.occurrences
+            for occurrence in occurrences:
+                index = int(occurrence[1])
+                rows[index] = po_entry
+        # ensure the number of final translations is same as the highest index
+        if rows:
+            assert len(rows) == int(max(rows.keys()))
+        # sort by index to have the expected order
+        return [po_entry for index, po_entry in sorted(rows.items())]
```

Comment:
Oooh, I wanna try out the new "suggestion" feature:
```suggestion
        return [po_entry for _, po_entry in sorted(rows.items())]
```
it's really a shame that list comprehension vars aren't scoped to the list comprehension.

Reply:
oh i updated it already and i cant apply it now since it has changed.